src/App.js

```diff
@@ -1,16 +1,35 @@
-import React from 'react';
+import React, {useState} from 'react';
 import './App.css';
 import chatMessages from './data/messages.json';
+import ChatLog from './components/ChatLog';
 
+
+
+
+
+  
 const App = () => {
-  return (
+  const [chatData, setChatData] = useState(chatMessages);
+  const handleToggleLiked = (updatedEntry) => {
+    const newChatData = chatData.map((entry) => {
+        if (entry.id === updatedEntry.id) {
+          return updatedEntry;
+        } else {
+          return entry
+        }
+  });
+setChatData(newChatData)}
+  const totalLikes = chatData.reduce((total, entry) => {
+    return total + entry.liked;
+}, 0);
```

Comment:
Since this logic is not in a method at first glance, it can be a little unclear when this should execute and what it's doing. To help organize your code, put this logic into a method like `getHeartCount` and then you can invoke the method in the JSX.


So line 29 would look like `<section id="totalLikes">{getHeartCount()} ❤️s</section>`